### PR TITLE
Changes to cuts for new interface

### LINKED ISSUE #51: MSlice will not recognise a cut when the parent is not in the ADS
A possible solution would be to reimplement CutAlgorith.is_cut and CutAlgorithm.get_cut_params without depending on the workspace history.
MSlice currently checks to see if a workspace is a cut by looping through the workspace history and searching for a call to BinMD.
MSlice extracts the parameters by looking at the properties passed to the BinMD algorithm.
This is superfluous and will fail if the parent algorithm does not exist in the ADS. 
To be exact 

```
        history = cut_workspace.getHistory()
        for i in range(history.size()-1,-1,-1):
            algorithm = history.getAlgorithm(i) #<- RuntimeError will be raised here 
```

The above snippet will fail if run on a workspace in ipython in mantid on workspace
 even though right-clicking the smae workspace and clicking show History will bring up the entire workspaces history

It is possible to deduct the cut axis specs by doing something along the lines of

```
dim = workspace.getDimension(0)
name = dim.getName()
min = dim.getMinimum()
max = dim.getMaximum()
step = dim.getX(1) - dim.getX(0)
```

All that is left is to find a way to find the integration limits without failing if parent has been removed from the ADS

### LINKED ISSUE #220: Slice / cut widget adjustments
Part of GUI redesign #204 

These widgets need some adjustments to work with the GUI changes:

- Automatically save cuts to workspace
- Possibly switch tabs automatically after cut/projection operations

### PR DESCRIPTION
**Pull request by MatthewBowles**
**Not to be merged until after #209**

- Add some workspace type specific controls to `main_window`
- Each cut is now saved as a workspace
- Some refactoring to `mslice.models.cut` to allow `MDHistoWorkspaces` to be loaded / plotted independently. Before, the cut parameters and parent workspace were cached and re-run every time the cut was plotted. Now, the data is saved and read from the `MDHistoWorkspace`.
- Changed name of cuts that are saved to a file to include integration limits -  before, they looked like `MAR21335_Ei60.00meV_QE_cut1` with an incrementing index on the end. Now they look like this: `MAR21335_Ei60_meV_QE_cut(-1.0,1.0) `
- Automatically change tabs after load/powder/plot cut operations

**To test:**
- Test all buttons that appear on each workspace tab function as expected
- save a cut to a file, then load it in a different session and plot (without a need for the parent workspace)

Fixes #51 
Fixes #220

### CODE REVIEW

**Comment by martyngigg** [comment]
@MatthewBowles There are a few conflicts in here now.

**Comment by martyngigg** [comment]
I have a question as to whether we have chose the best names for the workspace manager tabs. `2D`, `MDEvent` & `MDHisto` mean something to us but my concern is that this is mantid implementation language. I was wondering whether something like `Loaded`, `Slices` & `Cuts` might be more informative for users?
@mducle What do you think?

File: mslice/app/mainwindow.ui [hunk 1]
```diff
@@ -31,142 +31,228 @@
         <enum>QTabWidget::Rounded</enum>
        </property>
        <property name="currentIndex">
-        <number>0</number>
+        <number>1</number>
```
Comment (martyngigg) [inline mslice/app/mainwindow.ui]: Did we mean to change the default tab to the workspace manager?

Comment (MatthewBowles) [reply mslice/app/mainwindow.ui]: Nope, I wish QtDesigner wouldn't do this...

Comment (martyngigg) [reply mslice/app/mainwindow.ui]: I agree - it is very annoying.

**Comment by mducle** [comment]
@martyngigg @MatthewBowles I think for now the names are ok - the instrument scientists know what they mean. For the beta release we should change it though. I was thinking of either `Raw` or `Loaded` for `2D`; `Projected` for `MD Event` and `Rebinned` or `Saved` for `MD Histo`.

I'll add it to the things to discuss at the Mantid meeting.

**Comment by martyngigg** [comment]
@mducle Thanks.

**Pull request merged**